src/components/EventCalender/Yearly/YearlyEventCalender.tsx

```
@@ -158,106 +151,87 @@
               >
-                {expandedY === `${monthInx}-${dayIndex}` && renderedEvents}
+                {expandedY === expandKey && renderedEvents}
               </div>
-              {renderedEvents && renderedEvents?.length > 0 && (
+              {eventsForDate.length > 0 ? (
                 <button
                   className={styles.btn__more}
-                  onClick={() => toggleExpand(`${monthInx}-${dayIndex}`)}
+                  onClick={() => toggleExpand(expandKey)}
+                  data-testid={`expand-btn-${expandKey}`}
                 >
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)